ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextViewManager.java

```diff
@@ -58,18 +58,33 @@ public void setNumberOfLines(ReactTextView view, int numberOfLines) {
   @ReactProp(name = ViewProps.TEXT_ALIGN)
   public void setTextAlign(ReactTextView view, @Nullable String textAlign) {
     if (textAlign == null || "auto".equals(textAlign)) {
-      view.setGravity(Gravity.NO_GRAVITY);
+      view.setGravity(view.getGravity());
```

Comment:
Yes, this line could probably be removed, at least if NO_GRAVITY is the default behaviour.